### PR TITLE
Handle python versions in matrix.

### PR DESCRIPTION
**Pull request by cqc-alec**
As discussed [here](https://github.com/CQCL/tket/pull/529#discussion_r970694550).

The price we pay is storing the contents of `~/.conan/data/tket` in an artefact, which is then downloaded by each pytket build. But this is still a net gain in time, as well as a useful refactor.

I also factored out the script to set up the conan profile and options.

(Will require a small tweak if #540 is merged first.)

### CODE REVIEW

**Comment by cqc-alec** [comment]
Will modify "required" checks list after merge.

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -153,91 +148,79 @@ jobs:
...
+    - name: Test pytket with coverage
+      if: matrix.os == 'ubuntu-22.04' && matrix.pyver == '3.8'
       run: |
         cd pytket/tests
         pip install -r requirements.txt
         pytest --ignore=simulator/ --doctest-modules --cov=../pytket --cov-branch --cov-report=html --cov-report=xml:htmlcov/cov.xml
-    - name: Test pytket (3.8)
-      if: matrix.os != 'ubuntu-22.04' && github.event_name == 'push'
+    - name: Test pytket
+      if: matrix.os != 'ubuntu-22.04' || matrix.pyver != '3.8'
```
Comment (cqc-melf) [inline .github/workflows/build_and_test.yml:201]: Why are we doing the exclusion?

Comment (cqc-melf) [reply .github/workflows/build_and_test.yml:201]: is this the case where the coverage check is done?

Comment (cqc-alec) [reply .github/workflows/build_and_test.yml:201]: Yes, the tests are already run in that case using the coverage flags, so no need to run them again.

**Review by cqc-melf — APPROVED** [review]

**Pull request merged**